# SI 618 - Homework #1: Data Manipulation

Version 2021.02.04.01.CT

## Background

This homework assignment focuses on the analysis of historical data from the Olympic games.  The description of the data includes the following:
> This is a historical dataset on the modern Olympic Games, including all the Games from Athens 1896 to Rio 2016. I scraped this data from www.sports-reference.com in May 2018.

Your main task in this assignment is to explore the data *using the data
manipulation methods we covered in class* as well as those in the assigned readings.  You may need to consult pandas documentation, Stack Overflow, or other online resources.  


A total of 100 points is available in this homework assignment, consisting of:
- 80 points for completing the specific homework questions. More comprehensive 
answers will tend to gain more points.
- 10 points for the overall quality of spelling, grammar, punctuation, and style of written responses  (see [Strunk & White](https://drive.google.com/file/d/1L0P7xJwjUGBvyb49mL3dw1Bt7hzRTiTl/view) for detailed specifications).
- 10 points for creating code that conforms to [PEP 8](https://www.python.org/dev/peps/pep-0008/) guidelines.  You should review those guidelines before proceding with the assignment.


## Download the data from: 
- https://www.kaggle.com/heesoo37/120-years-of-olympic-history-athletes-and-results

<span style="color:magenta"> NOTE: You will only need the file named "athlete_events.csv" </span>

In [ ]:
MY_UNIQNAME = '' # please fill in your uniqname

In [ ]:
# Run but do not modify this cell
import numpy as np
MY_UNIQHASH = hash(MY_UNIQNAME) & 2**32-1
np.random.seed(MY_UNIQHASH)

## Answer the following questions.
Points for each question are shown below.

For each question, you should
1. Write code using Python and pandas that can help you answer the following questions, and
2. Explain your answers in plain English. You should use complete sentences that would be understood by an educated professional who is not necessarily a data scientist (like a product manager).

### <span style="color:magenta"> Q1: 5 pt: Describe the dataset. </span>
- How many number rows and columns do you see in each data frame? 
- Are there any other interesting descriptive details?

In [ ]:
# put your code here

(Use this space to explain your answers)

### <span style="color:magenta">Q2: 5 pt: Without using dataframe.describe(), calculate the median age, height, and weight for athletes in the data. Round your answer to the nearest hundredths. </span>

In [1]:
import pandas as pd

In [2]:
df = pd.DataFrame({'a':[1,2,3],'b':[4.123,1.222,1.0]})

In [3]:
df

a      b
0  1  4.123
1  2  1.222
2  3  1.000

In [6]:
import numpy as np

In [14]:
df.b.mean()

2.115

In [16]:
np.round(df.b.median(),2)

1.22

In [ ]:
# put your code here

The mean of ```b``` is 2.12

### <span style="color:magenta">Q3: 5 pt: According to the data, how many unique cities have hosted the Olympic games? How many cities have hosted multiple times? Which ones? </span>
You will likely want to use drop_duplicates() on some subset of your athlete dataframe.

In [ ]:
# put your code here

(Use this space to explain your answers)

### <span style="color:magenta">Q4: 10 pt: Generate histogram plots for the age, weight, and height for Danish female athletes who competed in the 1970s. Describe the plots and point out any interesting aspects. </span>

In [ ]:
# put your code here

(Use this space to explain your answers)

### <span style="color:magenta">Q5: 10 pt: Which athlete competed in the most number of events? In which events did the athelete participate, and for what range of years? Which country did the athlete represent? </span>

In [ ]:
# put your code here

(Use this space to explain your answers)

### <span style="color:magenta">Q6: 10 pt: How many gold, silver, and bronze medals did Sawao Kato, one of the most successful Olympic athletes of all time, win during his Olympic career? What is the breakdown of events and medal types won for each? </span>

In [ ]:
# put your code here

(Use this space to explain your answers)

### <span style="color:magenta">Q7: 10 pt: Which sport has the highest median athlete age? </span>
To determine this, you might want to plot the median age for each sport. You may want to sort the axes appropriately for your analysis.

In [ ]:
# put your code here

(Use this space to explain your answers)

### <span style="color:magenta">Q8: 10 pt: Which country won the most medal points in all of the games held in 1992? </span>
Rank each country by a score, where 3 points are assigned for a gold medal, 2 points for silver, and 1 point for bronze.

In [ ]:
# put your code here

(Use this space to explain your answers)

### <span style="color:magenta">Q9: 15 pt: The Soviet–American Cold War was a period of tension between the two superpowers between approximately the end of WWII and the late 1980s with the latent ideological tension playing out in a variety of ways including the Olympic Games. Which of the two countries won the most medals during the Cold War? </span>

Note that to answer this question you will need to determine reasonable values for the beginning and end years of the Cold War. There is no single correct answer and you should justify your choices, citing your sources as appropriate.

In [ ]:
# put your code here

(Use this space to explain your answers)

## Please submit your completed notebook in .IPYNB and .HTML formats via Canvas

In [34]:
df = pd.DataFrame({'medal':['Bronze','Silver','Gold','Bronze','Bronze'],'country':['USA','USA','CAN','CAN','AUS']})

In [35]:
df

medal country
0  Bronze     USA
1  Silver     USA
2    Gold     CAN
3  Bronze     CAN
4  Bronze     AUS

In [36]:
points = {'Gold':3, 'Silver':2, 'Bronze':1}

In [37]:
points

{'Gold': 3, 'Silver': 2, 'Bronze': 1}

In [38]:
df['points'] = df['medal'].map(points)

In [39]:
df

medal country  points
0  Bronze     USA       1
1  Silver     USA       2
2    Gold     CAN       3
3  Bronze     CAN       1
4  Bronze     AUS       1

In [40]:
df.groupby('country')['points'].sum()

country
AUS    1
CAN    4
USA    3
Name: points, dtype: int64

In [41]:
df

medal country  points
0  Bronze     USA       1
1  Silver     USA       2
2    Gold     CAN       3
3  Bronze     CAN       1
4  Bronze     AUS       1

In [62]:
df.country.value_counts() # this is a series

CAN    2
USA    2
AUS    1
Name: country, dtype: int64

In [58]:
dictionary_of_value_counts = {'counts':the_counts} # this is a dictionary

In [ ]:
df3 = pd.DataFrame({'counts':df.country.value_counts()})

In [59]:
df2 = pd.DataFrame(dictionary_of_value_counts) # this creates a new dataframe

In [60]:
df2

counts
CAN       2
USA       2
AUS       1

In [51]:
df2[df2.counts > 1]

counts
CAN       2
USA       2